pxf/pxf-json/src/main/java/org/apache/hawq/pxf/plugins/json/JsonResolver.java

```diff
@@ -0,0 +1,259 @@
+
+				// Get the JSON projections from the column name
+				// For example, "user.name" turns into ["user","name"]
+				String[] projs = cd.columnName().split("\\.");
+
+				// Move down the JSON path to the final name
+				JsonNode node = getPriorJsonNode(root, projs);
+
+				// If this column is an array index, ex. "tweet.hashtags[0]"
+				if (isArrayIndex(projs)) {
+
+					// Get the node name and index
+					String nodeName = getArrayName(projs);
+					int arrayIndex = getArrayIndex(projs);
+
+					// Move to the array node
+					node = node.get(nodeName);
+
+					// If this node is null or missing, add a null value here
+					if (node == null || node.isMissingNode()) {
+						addNullField(columnType);
+					} else if (node.isArray()) {
+						// If the JSON node is an array, then add it to our list
+						addFieldFromJsonArray(columnType, node, arrayIndex);
+					} else {
+						throw new InvalidParameterException(nodeName + " is not an array node");
+					}
+				} else {
+					// This column is not an array type
+					// Move to the final node
+					node = node.get(projs[projs.length - 1]);
+
+					// If this node is null or missing, add a null value here
+					if (node == null || node.isMissingNode()) {
+						addNullField(columnType);
+					} else {
+						// Else, add the value to the record
+						addFieldFromJsonNode(columnType, node);
+					}
+				}
+			}
+		}
+
+		return list;
+	}
+
+	/**
+	 * Iterates down the root node to the prior JSON node. This node is used
+	 * 
+	 * @param root
+	 * @param projs
+	 * @return
+	 */
+	private JsonNode getPriorJsonNode(JsonNode root, String[] projs) {
+
+		// Iterate through all the tokens to the desired JSON node
+		JsonNode node = root;
+		for (int j = 0; j < projs.length - 1; ++j) {
+			node = node.path(projs[j]);
+		}
+
+		return node;
+	}
+
+	/**
+	 * Gets a boolean value indicating if this column is an array index column
+	 * 
+	 * @param projs
+	 *            The array of JSON projections
+	 * @throws ArrayIndexOutOfBoundsException
+	 */
+	private boolean isArrayIndex(String[] projs) {
+		return projs[projs.length - 1].contains("[") && projs[projs.length - 1].contains("]");
+	}
+
+	/**
+	 * Gets the node name from the given String array of JSON projections, parsed from the ColumnDescriptor's
+	 * 
+	 * @param projs
+	 *            The array of JSON projections
+	 * @return The name
+	 * @throws ArrayIndexOutOfBoundsException
+	 */
+	private String getArrayName(String[] projs) {
+		return projs[projs.length - 1].replaceAll("\\[[0-9]+\\]", "");
+	}
+
+	/**
+	 * Gets the array index from the given String array of JSON projections, parsed from the ColumnDescriptor's name
+	 * 
+	 * @param projs
+	 *            The array of JSON projections
+	 * @return The index
+	 * @throws ArrayIndexOutOfBoundsException
+	 */
+	private int getArrayIndex(String[] projs) {
+		return Integer.parseInt(projs[projs.length - 1].substring(projs[projs.length - 1].indexOf('[') + 1,
```

Comment:
perhaps use a variable for `String arrayElem = projs[projs.length - 1]` - will make the code shorter.